homeassistant/components/openexchangerates/config_flow.py

```diff
@@ -0,0 +1,155 @@
+"""Config flow for Open Exchange Rates integration."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Mapping
+from typing import Any
+
+from aioopenexchangerates import (
+    Client,
+    OpenExchangeRatesAuthError,
+    OpenExchangeRatesClientError,
+)
+import async_timeout
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_BASE
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CLIENT_TIMEOUT, DEFAULT_BASE, DOMAIN, LOGGER
+
+
+def get_data_schema(
+    currencies: dict[str, str], existing_data: Mapping[str, str]
+) -> vol.Schema:
+    """Return a form schema."""
+    return vol.Schema(
+        {
+            vol.Required(CONF_API_KEY): str,
+            vol.Optional(
+                CONF_BASE, default=existing_data.get(CONF_BASE) or DEFAULT_BASE
+            ): vol.In(currencies),
+        }
+    )
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, str]) -> dict[str, str]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    client = Client(data[CONF_API_KEY], async_get_clientsession(hass))
+
+    try:
+        async with async_timeout.timeout(CLIENT_TIMEOUT):
+            await client.get_latest(base=data[CONF_BASE])
+    except OpenExchangeRatesAuthError as err:
+        raise InvalidAuth from err
+    except OpenExchangeRatesClientError as err:
+        raise CannotConnect from err
```

Comment:
Personal preference, but I'd probably let the existing typed exceptions raise and trap them below instead of wrapping them again.